seata-spring-boot-starter/src/main/java/io/seata/spring/boot/autoconfigure/provider/SpringBootConfigurationProvider.java

```diff
@@ -139,44 +79,71 @@ public Object intercept(Object proxy, Method method, Object[] args, MethodProxy
         });
     }
 
-    private Object get(String dataId, Object defaultValue, long timeoutMills) throws IllegalAccessException, ExecutionException, InterruptedException, TimeoutException {
+    private Object get(String dataId, Object defaultValue, long timeoutMills) throws IllegalAccessException, InstantiationException {
         return get(dataId, defaultValue);
 
     }
 
-    private Object get(String dataId, Object defaultValue) throws IllegalAccessException, ExecutionException, InterruptedException, TimeoutException {
+    private Object get(String dataId, Object defaultValue) throws IllegalAccessException, InstantiationException {
         Object result = get(dataId);
         if (result == null) {
             return defaultValue;
         }
         return result;
     }
 
-    private Object get(String dataId) throws IllegalAccessException, ExecutionException, InterruptedException, TimeoutException {
+    private Object get(String dataId) throws IllegalAccessException, InstantiationException {
         String propertyPrefix = getPropertyPrefix(dataId);
-        CompletableFuture<Object> propertyFuture = PROPERTY_BEAN_MAP.get(propertyPrefix);
-        if (propertyFuture != null) {
-            Object propertyObject = propertyFuture.get(10000, TimeUnit.MILLISECONDS);
-            String propertySuffix = getPropertySuffix(dataId);
-            Optional<Field> fieldOptional = Stream.of(propertyObject.getClass().getDeclaredFields()).filter(
-                f -> f.getName().equalsIgnoreCase(propertySuffix)).findAny();
-            if (fieldOptional.isPresent()) {
-                Field field = fieldOptional.get();
-                field.setAccessible(true);
-                Object valueObject = field.get(propertyObject);
-                if (valueObject instanceof Map) {
-                    String key = StringUtils.substringAfterLast(dataId, String.valueOf(DOT));
-                    valueObject = ((Map) valueObject).get(key);
-                }
-                return valueObject;
+        String propertySuffix = getPropertySuffix(dataId);
+        ApplicationContext applicationContext = (ApplicationContext) ObjectHolder.INSTANCE.getObject(OBJECT_KEY_SPRING_APPLICATION_CONTEXT);
+        Class<?> propertyClass = PROPERTY_BEAN_MAP.get(propertyPrefix);
+        Object valueObject = null;
+        if (propertyClass != null) {
+            try {
+                Object propertyBean = applicationContext.getBean(propertyClass);
+                valueObject = getFieldValue(propertyBean, propertySuffix, dataId);
```

Comment:
when the bean exists, the corresponding value already exists in spring env, why do you need to get the value through reflection?

Reply:
Because we don't know which method of which class will be called at encoding time, we need reflection to get the corresponding value. Otherwise, we need to check the key, such as the If (propertyClass = = someClass && filedName = = someFieldName) {propertyClass. GetFileName ()}. There will be many such checks that need to be written in advance.